src/crafting.cpp

```diff
@@ -387,22 +116,24 @@ void player::long_craft()
 
 bool player::making_would_work( const std::string &id_to_make, int batch_size )
 {
-    const recipe *making = recipe_by_name( id_to_make );
-    if( making == nullptr || !crafting_allowed( *this, *making ) ) {
+    const auto &making = recipe_dict[ id_to_make ];
+    if( !( making && crafting_allowed( *this, making ) ) ) {
         return false;
     }
 
-    if( !can_make( making, batch_size ) ) {
+    if( !can_make( &making, batch_size ) ) {
         std::ostringstream buffer;
         buffer << _( "You can no longer make that craft!" ) << "\n";
-        buffer << making->requirements().list_missing();
+        buffer << making.requirements().list_missing();
         popup( buffer.str(), PF_NONE );
         return false;
     }
 
-    return making->check_eligible_containers_for_crafting( batch_size );
+    return making.check_eligible_containers_for_crafting( batch_size );
 }
 
+recipe::recipe() : skill_used( NULL_ID ) {}
```

Comment:
Since some recipes (notably `null_recipe` in recipe_dictionary.cpp) are initialized in static context, this `NULL_ID` may cause problems depending on compiler. This dependence is known to cause builds for MacOSX to segfault at start.